WordPress/openverse-frontend
# Fix CSS import ordering

## Fixes


Fixes #992 by @zackkrida 

## Description


Using the [templates](https://nuxtjs.org/docs/configuration-glossary/configuration-build#templates) option in Nuxt config we can override the problematic templates to fix the import order. I've opened a PR upstream to address this but it hasn't gotten any attention from maintainers: https://github.com/nuxt/nuxt.js/pull/10275

The idea here is to ensure that the CSS imports happen before any other styles are defined, especially `@apply` Tailwind styles.

We shouldn't stop investigating ways of fixing this after this PR; maintaining these template overrides is far from ideal. I wasn't able to reproduce the behavior we're seeing in our application [in codesandbox](https://codesandbox.io/s/eager-glade-gewem) so I feel like there's something very subtly wrong with our configuration that is somehow causing this CSS misordering to occur. I just don't know what it could be and we need a stop-gap for this problem in the meantime.

## Testing Instructions

Checkout `main` and run `pnpm build` and take note of the existing import ordering in `./.nuxt/index.js` and `./.nuxt/App.js`.

Now checkout this branch and run `pnpm build` and verify that the overridden templates are applied in the same files. Compare them to the previous versions to confirm that they've changed _only in the ways described in the README.md added in this PR_.

Run `pnpm dev` for good measure and make sure the template changes are also present.

Next try to reproduce the bug in the linked issue, verify that it is fixed. Also verify that there are no more "ghost buttons" that we were seeing previously (like the skip to content button, which can be accessed by pressing `tab` immediately after loading a search results page).

This fix also introduced a regression of the double focus rings on the search type switcher; turns out we were fighting windmills there too. Verify that these buttons have a single focus ring. Also test the page menu on desktop and mobile widths.

Please test in Firefox, Chrome and Safari if possible and both with desktop and mobile viewport widths, with and without keyboard navigation.

Lastly, smoke test the app for styles and make sure that things "make sense" and that nothing is visually broken (in particular Buttons as that was the component that was being imported "too early"). I'd suggest using keyboard navigation to highlight all the focus rings and such around the application and make sure they still work.

## Checklist

- [N/A] I added or updated tests for the changes I made (if applicable).

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
